README.md

```diff
@@ -69,51 +72,112 @@ Build using the `Makefile`
+Ability to retrieve SBOM files from URLs over HTTPS, OCI, or Git.
 
-TBD
+```shell
+bomctl fetch [flags] SBOM_URL...
 
-### List (Planned)
+Flags:
+  -h, --help               help for fetch
+      --netrc              Use .netrc file for authentication to remote hosts
+  -o, --output-file FILE   Path to output file
+```
 
-TBD
+This includes recursive loading of external references in an SBOM to other SBOMs and placing them into the persistent cache. If SBOMs are access controlled, a user's [.netrc](https://www.gnu.org/software/inetutils/manual/html_node/The-_002enetrc-file.html) file to authenticate.
 
-### Merge (Planned)
+The following example will fetch this [CycloneDX SBOM](https://raw.githubusercontent.com/bomctl/bomctl-playground/main/examples/bomctl-container-image/bomctl_bomctl_v0.3.0.cdx.json)
+that represents a container image and will then recursively fetch an externally referenced SBOM that
+represents an application within the container image.
 
-TBD
+```shell
+bomctl fetch https://raw.githubusercontent.com/bomctl/bomctl-playground/main/examples/bomctl-container-image/bomctl_bomctl_v0.3.0.cdx.json
+```
 
-### Push (Planned)
+### Import
 
-TBD
+Import SBOM files from either standard input or the local file system.
 
-### Redact (Planned)
+```shell
+bomctl import [flags] { - | FILE...}
 
-TBD
+Flags:
+  -h, --help   help for import
+```
 
-### Split (Planned)
+### List
 
-TBD
+List cached SBOM documents.
 
-### Trim (Planned)
+```shell
+bomctl list [flags] SBOM_ID...
+
+Aliases:
+  list, ls
+
+Flags:
+  -h, --help   help for list
+```
 
-TBD
+## Roadmap
+
+The project is focused on building an architecture that enables reading in, operating on, and reading
+out collections of SBOM files independent of format. This project requires an architecture that prioritizes the relationships between components in a SBOM document and between SBOM documents.
+
+Once this architecture is established, more complex operations can be implemented. This operations include:
+
+- `Diff`
+  - Generating diffs between components and component dependencies
+  - Generating diffs of component properties
+  - Creating diffs in machine readable and human readable formats
+- `Enrich`
+  - Use the [Transparency Exchange API](https://github.com/CycloneDX/transparency-exchange-api) to find enrich sbom components with additional data.
+  - Interface with [GUAC](https://guac.sh/)
+- `Merge`
+  - Merge fields from similar components in two or more SBOM documents
+  - Merge components and component dependencies from two or more SBOM documents
+  - Flatten multiple SBOM document component dependencies into a single SBOM document
+- `Redact`
+  - Redact fields by regular express or field name while keeping tracibility to the original document
+- `Split`
+  - Split an SBOM dependency tree into multiple files based on purl type or component identifier
+- `Trim`
+  - Trim an SBOM dependency tree based on purl type or component identifier
+- __Your suggestions too!__
+
```

Comment:
Mention previous architecture decisions and where they can be found and maybe link to a search for ADR issues that were closed